src/web/components/Core/PagingTool.tsx

```diff
@@ -0,0 +1,160 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useEffect, useState } from 'react';
+
+import { SelectDropdown, SelectOption } from './SelectDropdown';
+
+import './PagingTool.scss';
+
+const rowsPerPageValues = [10, 25, 50, 100, 250] as const;
+
+export type RowsPerPageValues = (typeof rowsPerPageValues)[number];
+
+type PagingProps = Readonly<{
+  numberTotalRows: number;
+  onChangeRows: (currentPageNumber: number, currentRowsPerPage: RowsPerPageValues) => void;
+  rowsPerPageTitle?: string;
+  initialRowsPerPage?: RowsPerPageValues;
+  initialPageNumber?: number;
+}>;
+
+export function PagingTool({
+  numberTotalRows,
+  onChangeRows,
+  rowsPerPageTitle = 'Rows Per Page',
+  initialRowsPerPage = 10,
+  initialPageNumber = 1,
+}: PagingProps) {
+  const initialPageNumberOptions = Array.from(
+    { length: Math.ceil(numberTotalRows / initialRowsPerPage) },
```

Comment:
There's a bunch of math in this component which might be nice to be refactored out. As an example, this math is done twice in this file. Then we could potentially add tests around them.

Ideally this file should probably only change if the wanted to change how it looked, while the behind-the-scenes calculations could be extracted away into a separate `.ts` file.

Does that seem reasonable?

Reply:
Definitely - it cleaned up the file nicely